File: Civitas_TUI/src/JuegoTexto/Controlador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package JuegoTexto;

import java.util.ArrayList;
import civitas.*;

/**
 *
 * @author alejandro
 */
public class Controlador {
    private CivitasJuego juego;
    private VistaTextual vista;
    
    public Controlador(CivitasJuego jue, VistaTextual vist){
        juego = jue;
        vista = vist;
    }
    
    public void juega(){
        vista.setCivitasJuego(juego);
        boolean prueba = juego.finalDelJuego();
        while(!juego.finalDelJuego()){
            vista.actualizarVista();
            vista.pausa();
            OperacionesJuego operacion = vista.juegoModel.siguientePaso();
            vista.mostrarSiguienteOperacion(operacion);
            if(operacion != OperacionesJuego.PASAR_TURNO){
                vista.mostrarEventos();
            }
            if(!vista.juegoModel.finalDelJuego()){
                switch(operacion){
                    case COMPRAR:
                        Casilla_Calle calle = (Casilla_Calle) vista.juegoModel.getCasillaActual();
                        if(!calle.getTituloPropiedad().tienePropietario()){
                            Respuestas respuesta = vista.comprar();
                            if(respuesta == Respuestas.SI)
                                vista.juegoModel.comprar();
                        }
                        vista.juegoModel.siguientePasoCompletado(operacion);
                    break;
                    
                    case GESTIONAR:
                        vista.gestionar();
                        if(vista.iGestion != 5){
                            OperacionInmobiliaria gestion = new OperacionInmobiliaria(GestionesInmobiliarias.values()[vista.iGestion], vista.iPropiedad);
                            switch(gestion.getGestion()){
                                case VENDER:
                                    vista.juegoModel.vender(gestion.getNumPropiedad());
                                break;
                                
                                case HIPOTECAR:
                                    vista.juegoModel.hipotecar(gestion.getNumPropiedad());
                                break;
                                
                                case CANCELAR_HIPOTECA:
                                    vista.juegoModel.cancelarHipoteca(gestion.getNumPropiedad());
                                break;
                                
                                case CONSTRUIR_CASA:
                                    vista.juegoModel.construirCasa(gestion.getNumPropiedad());
                                break;
                                
                                case CONSTRUIR_HOTEL:
                                    vista.juegoModel.construirHotel(gestion.getNumPropiedad());
                                break;
                            }
                        }
                        
                        else{
                            vista.juegoModel.siguientePasoCompletado(operacion);
                        }
                    break;
                    
                    case SALIR_CARCEL:
                        SalidasCarcel metodo = vista.salirCarcel();
                        
                        if(metodo == SalidasCarcel.PAGANDO)
                            vista.juegoModel.salirCarcelPagando();
                        else
                            vista.juegoModel.salirCarcelTirando();
                    break;
                }
            }
        }
        
        ArrayList<Jugador> clasificacion = vista.juegoModel.ranking();
        
        System.out.println("\nCLASIFICACIÓN FINAL");
        
        for(int i = clasificacion.size()-1; i >= 0; i--)
            System.out.println(clasificacion.get(i));
    }
}

File: Civitas_Ruby/lib/controlador.rb
#encoding:utf-8
require_relative 'operaciones_juego.rb'
require_relative 'gestiones_inmobiliarias.rb'
require_relative 'civitas_juego.rb'
require_relative 'jugador.rb'
require_relative 'titulo_propiedad.rb'
require_relative 'civitas_juego.rb'
require_relative 'operacion_inmobiliaria.rb'
require_relative 'gestiones_inmobiliarias.rb'
require_relative 'diario.rb'
require_relative 'salidas_carcel.rb'
require_relative 'vista_textual.rb'
module Civitas
  class Controlador
    def initialize(jue, vist)
      @juego = jue
      @vista = vist
    end
    
    def juega()
      @vista.setCivitasJuego(@juego)
      prueba = @juego.finalDelJuego()
      
      while(!@juego.finalDelJuego())
        
        @vista.actualizarVista()
        @vista.pausa()
        
        operacion = @vista.juegoModel.siguientePaso()
        
        @vista.mostrarSiguienteOperacion(operacion)
        
        if(operacion != Operaciones_juego::PASAR_TURNO)
          @vista.mostrarEventos()
        end
        
        if(!@vista.juegoModel.finalDelJuego())
          case operacion
          when Operaciones_juego::COMPRAR
            #comprabacion de que ademas se deba de comprar que sea una calle para saber si se puede comprar
            if(@vista.juegoModel.getCasillaActual().tituloPropiedad != nil)
              if(!@vista.juegoModel.getCasillaActual().tituloPropiedad.tienePropietario())
                respuesta = @vista.comprar()
                if (respuesta == Respuestas::SI)
                  @vista.juegoModel.comprar()
                end
              end
            end
            
            #comprobacion de la casilla impuesto, la debemos de hacer aqui?
            puts "\n El siguinte paso es #{operacion}\n"
            @vista.juegoModel.siguientePasoCompletado(operacion)
            
            
          when Operaciones_juego::GESTIONAR
            @vista.gestionar()
            if(@vista.iGestion != 5)
              operaciones = [GestionesInmobiliarias::VENDER, GestionesInmobiliarias::HIPOTECAR, GestionesInmobiliarias::CANCELAR_HIPOTECA, GestionesInmobiliarias::CONSTRUIR_CASA, GestionesInmobiliarias::CONSTRUIR_HOTEL, GestionesInmobiliarias::TERMINAR]
              op_inmobiliaria = OperacionInmobiliaria.new(operaciones[@vista.iGestion],@vista.iPropiedad)
              
              case op_inmobiliaria.gestion
              when GestionesInmobiliarias::VENDER
                @vista.juegoModel.vender(op_inmobiliaria.numPropiedad)
              when GestionesInmobiliarias::HIPOTECAR
                @vista.juegoModel.hipotecar(op_inmobiliaria.numPropiedad)
              when GestionesInmobiliarias::CANCELAR_HIPOTECA
                @vista.juegoModel.cancelarHipoteca(op_inmobiliaria.numPropiedad)
              when GestionesInmobiliarias::CONSTRUIR_CASA
                @vista.juegoModel.construirCasa(op_inmobiliaria.numPropiedad)
              when GestionesInmobiliarias::CONSTRUIR_HOTEL
                @vista.juegoModel.construirHotel(op_inmobiliaria.numPropiedad)
              end
              
            else
              @vista.juegoModel.siguientePasoCompletado(operacion)
            end
            
            
          when Operaciones_juego::SALIR_CARCEL
            metodo = @vista.salirCarcel()
            if(metodo == SalidasCarcel::PAGANDO)
              exito = @vista.juegoModel.salirCarcelPagando()
            else
              exito = @vista.juegoModel.salirCarcelTirando()
            end
            
            if(!exito)
              @vista.juegoModel.siguientePasoCompletado(operacion)
            end
            
          end
        end
        
      end
      
      clasificacion = @vista.juegoModel.ranking()
    
      puts "\nCLASIFICACION FINAL"
      
      i = clasificacion.length()-1
      
      while i > -1
        puts clasificacion[i]
        i = i -1
      end
    end
    
    
    
    
  end
end